### PR TITLE
Adding support for Azure IoT Hub

### PR DESCRIPTION
**Pull request by ksaye**
Simple testing documentation here: https://kevinsaye.wordpress.com/2021/04/26/using-tasmota-with-azure-iot-hub/ until I can contribute tot the Tasmota documentation site.

## Description:

**Related issue (if applicable):** fixes #<Tasmota issue number goes here>

## Checklist:
  - [X] The pull request is done against the latest development branch
  - [X] Only relevant files were touched
  - [X] Only one feature/fix was added per PR and the code change compiles without warnings
  - [X] The code change is tested and works on Tasmota core ESP8266 V.2.7.4.9
  - [ ] The code change is tested and works with core ESP32 V.1.0.6
  - [X] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

_NOTE: The code change must pass CI tests. **Your PR cannot be merged unless tests pass**_

### CODE REVIEW

**Comment by ksaye** [comment]
There is a free SKU for Azure IoT Hub, but if a tester wants access to an IoT Hub an you do not want to create an Azure subscription, message me directly and I will create an IoT Hub for you.

**Comment by ascillato** [comment]
Thanks. 👍 

The link to the documentation you did is very complete and, if you want, it can be added to the tasmota´s docs.

Just a detail, the CLIENT NAME must be unique for each device. That should be remarked as it is a very common mistake.
Also, the link to the pull request in your documentation is the old PR and not this one.

File: tasmota/xdrv_02_mqtt.ino [hunk 1]
```diff
@@ -225,7 +231,14 @@ void MqttDisconnect(void) {
 }
 
 void MqttSubscribeLib(const char *topic) {
-  MqttClient.subscribe(topic);
+  #if defined(USE_MQTT_AZURE_IOT)
+    // Azure IoT Hub currently does not support custom topics: https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-mqtt-support
+    String realTopicString = "devices/" + String(SettingsText(SET_MQTT_CLIENT));
+    realTopicString += "/messages/devicebound/#";
+    MqttClient.subscribe(realTopicString.c_str());
+  #else //USE_MQTT_AZURE_IOT
```
Comment (s-hadinger) [inline tasmota/xdrv_02_mqtt.ino:239]: I'm ok with the code, however identation does not follow the Tasmota convention. The ifdefs start at the beginning of the line, and the code is identated as usual.

Comment (arendst) [reply tasmota/xdrv_02_mqtt.ino:239]: Agree. I will fix this once merged.

**Comment by ksaye** [comment]
@arendst from a documentation perspective, I plan on writing a better document, once I get 2 more changes in.  

(Perhaps a discussion item) Can the area 'Smart Home Integrations' be renamed to something else?  AWS and Azure (and other solutions) are not always Smart Home and have other uses.  I have a customer using Tasmota for a people counter in a corporate environment -- which is why I submitted the Azure IoT Hub code.

**Comment by arendst** [comment]
@ksaye regarding documentation pls contact the document site directly https://tasmota.github.io/docs/

@blakadder FYI

**Pull request merged**